File: dailyques/Q1390.cpp
//Q1390.cpp
class Solution {
public:
    int sumFourDivisors(vector<int>& nums) {
        int ans = 0;
        for (int x : nums) {
            int s = 1 + x, cnt = 2;
            for (int d = 2; d * d <= x; d++) {
                if (x % d == 0) {
                    int e = x / d;
                    cnt += (d == e ? 1 : 2);
                    s += (d == e ? d : d + e);
                    if (cnt > 4) break;
                }
            }
            if (cnt == 4) ans += s;
        }
        return ans;
    }
};

File: dailyques/Q1390.py
class Solution:
    def sumFourDivisors(self, nums):
        ans = 0
        for x in nums:
            s, cnt = 1 + x, 2
            d = 2
            while d * d <= x:
                if x % d == 0:
                    e = x // d
                    cnt += 1 if d == e else 2
                    s += d if d == e else d + e
                    if cnt > 4: break
                d += 1
            if cnt == 4: ans += s
        return ans
